rpc/qubesbuilder.PESign

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -e -o pipefail
+
+CERTIFICATE="$1"
+[[ -z "$CERTIFICATE" ]] && { echo "Please provide certificate name" >&2; exit 1; };
+
+PESIGN_ARGS=()
+# support per-certificate extra args, needed for hardware tokens
+if [ -e "$HOME/.config/qubes-pesign/${CERTIFICATE}" ];
+    # this may also override CERTIFICATE
+    . "$HOME/.config/qubes-pesign/${CERTIFICATE}"
+else
+    # just demangle name
+    CERTIFICATE="${CERTIFICATE//__/ }"
+fi
+
+PAYLOAD_DIR="$(mktemp -d)"
+
+cleanup() {
+    local payload_dir="$1"
+    if [ -n "${payload_dir}" ]; then
+        rm -rf "${payload_dir}"
+    fi
+}
+
+# shellcheck disable=SC2064
+trap "cleanup ${PAYLOAD_DIR}" EXIT
+
+payload="${PAYLOAD_DIR}/payload"
+
+# Limit stdin size
+head --bytes=100MB > "$payload"
+
+# We don't allow payload being at least 100MiB
+if [ "$(stat --format=%s "$payload")" -ge $((100 * 1024 * 1024)) ]; then
```

Comment:
```suggestion
size=$(stat --format=%s "$payload")
if [ "$size" -ge "$((100 * 1024 * 1024))" ]; then
```

Also please include a call to the PE checker I wrote.

Reply:
link?

Reply:
https://github.com/QubesOS/qubes-pecheck